blas/src/KokkosBlas1_iamax.hpp

```diff
@@ -49,13 +49,14 @@ typename XVector::size_type iamax(const XVector& x) {
       typename XVector::device_type, Kokkos::MemoryTraits<Kokkos::Unmanaged> >
       XVector_Internal;
 
-  typedef Kokkos::View<index_type, typename XVector_Internal::array_layout,
-                       Kokkos::HostSpace,
+  using layout_t = typename XVector_Internal::array_layout;
+
+  typedef Kokkos::View<index_type, layout_t, Kokkos::HostSpace,
                        Kokkos::MemoryTraits<Kokkos::Unmanaged> >
       RVector_Internal;
 
   index_type result;
-  RVector_Internal R = RVector_Internal(&result);
+  RVector_Internal R = RVector_Internal(&result, layout_t());
```

Comment:
Don't we want to initialize the layout with appropriate values?
Basically the same as `x` but with the first extent set to 1?

So if x has extents `nrows` and `nvecs` we end up with R having extents `1` and `nvecs`?

Reply:
For this overload of iamax, XVector can only be rank-1 and result is a single index, so R is always rank-0.